projects/packages/videopress/src/client/lib/url/index.ts

```diff
@@ -129,3 +129,44 @@ export function buildVideoPressURL(
 export const removeFileNameExtension = ( name: string ) => {
 	return name.replace( /\.[^/.]+$/, '' );
 };
+
+/**
+ * Helper function to create and return textarea element.
+ *
+ * > Moreover, using textContent can prevent XSS attacks.
+ * https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#differences_from_innerhtml
+ *
+ * It will be used to decode HTML entities,
+ * As long as element doesn’t get inserted in the DOM,
+ * we’re good in terms of security,
+ * since textContent will return the content without evaluating it.
+ *
+ * @returns {HTMLTextAreaElement} - Textarea element
+ */
+const createTextareaElement = (): HTMLTextAreaElement => {
+	if ( document.implementation && document.implementation.createHTMLDocument ) {
+		return document.implementation.createHTMLDocument( '' ).createElement( 'textarea' );
+	}
+
+	return document.createElement( 'textarea' );
+};
+
+/**
+ * Decode the given text, replacing HTML entities
+ * with their corresponding characters.
+ *
+ * @param {string} text     - Text to decode
+ * @returns {string | null} Decoded text
+ */
+export function decodeEntities( text: string ): string | null {
+	if ( ! text?.length ) {
```

Comment:
As per the types, we expect `text` to be a string. In that case, is there a scenario where `.length` won't exist? I don't think so, so we should be able to remove the optional chaining:

```suggestion
	if ( ! text.length ) {
```